zerver/actions/message_edit.py

```diff
@@ -363,7 +363,7 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
             "flags": um.flags_list(),
         }
 
-    send_event(user_profile.realm, event, list(map(user_info, ums)))
+    send_event_on_commit(user_profile.realm, event, list(map(user_info, ums)))
```

Comment:
This looks good, though I do notice that the caller does `render_incoming_message` inside a transaction; maybe we should discuss that detail in #backend, as I'm not sure that's a good idea.

Reply:
To change it, I think we'd need to first fetch the message without `select_for_update`, and then have a shorter transaction that just does `select_for_update` followed immediately by `do_update_embedded_data` ... which might be mostly cleanly done by just moving the `@transaction.atomic` to decorating this function, and having this function do a reselect-for-update to lock the message? Maybe you can just do that and post in #backend for visibility.